Rules/CommonScripts/KillMessages.as

```diff
@@ -288,11 +291,48 @@ void onPlayerDie(CRules@ this, CPlayer@ victim, CPlayer@ killer, u8 customdata)
 			CBlob@ victimblob = victim.getBlob();
 			CPlayer@ helper = getAssistPlayer(victim, killer);
 
-			bool kill = killerblob !is null && victimblob !is null && killerblob.isMyPlayer() && killerblob !is victimblob;
+			bool kill = killerblob !is null && victimblob !is null && killerblob !is victimblob;
 			bool assist = helper !is null && helper.isMyPlayer();
+
 			if (kill)
 			{
-				add_message(KillSpreeMessage(victim));
+				if (isServer())
+				{
+					if (killerblob.getTeamNum() != victimblob.getTeamNum())
+					{
+						killer.set_u32("kill time", getGameTime());
+						
+						if(!killer.exists("killstreak"))
+						{
+							killer.set_u8("killstreak", 1);
+						}
+						else
+						{
+							killer.add_u8("killstreak", 1);
```

Comment:
I'm pretty sure `add_u8` adds from zero if `"killstreak"` doesn't exist. If this is the case, the surrounding `if else` statement becomes redundant. I may be wrong though.

Reply:
I'll keep it to make the intention clear